src/EditorFeatures/CSharpTest/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -361,6 +361,56 @@ int Goo
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        [WorkItem(29746, "https://github.com/dotnet/roslyn/issues/29746")]
+        public async Task FieldReturnedInMethod()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private string [|_s|];
+    public MyClass(string s) => _s = s;
+    public string Method()
+    {
+        return _s;
+    }
+}",
+@"class MyClass
+{
+    private readonly string [|_s|];
+    public MyClass(string s) => _s = s;
+    public string Method()
+    {
+        return _s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        [WorkItem(29746, "https://github.com/dotnet/roslyn/issues/29746")]
+        public async Task FieldReadInMethod()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private string [|_s|];
+    public MyClass(string s) => _s = s;
+    public string Method()
+    {
+        return _s.ToUpper();
+    }
+}",
+@"class MyClass
+{
+    private readonly string [|_s|];
+    public MyClass(string s) => _s = s;
+    public string Method()
+    {
+        return _s.ToUpper();
```

Comment:
Maybe it would be good to add a test where _s is a struct (where the method could modify it, therefore can't be readonly) and with a readonly struct where it should be offered again?

Reply:
This is intended to be covered by `FieldTypeIsMutableStruct` and `FieldTypeIsCustomImmutableStruct`.